lib/replica_pools/query_cache.rb

```diff
@@ -18,29 +18,20 @@ module QueryCache
     # select_all is trickier. it needs to use the leader
     # connection for cache logic, but ultimately pass its query
     # through to whatever connection is current.
-    def select_all(*args, **kwargs)
-      relation, name, raw_binds = args
-
-      if !query_cache_enabled || locked?(relation)
-        return route_to(current, :select_all, *args, **kwargs)
+    def select_all(arel, name = nil, binds = [], *args, preparable: nil, **kwargs)
+      if !query_cache_enabled || locked?(arel)
+        return route_to(current, :select_all, arel, name, binds, *args, preparable: preparable, **kwargs)
       end
 
-      # duplicate binds_from_relation behavior introduced in 4.2.
-      if raw_binds.blank? && relation.is_a?(ActiveRecord::Relation)
-        arel, binds = relation.arel, relation.bind_values
```

Comment:
This was the line that was erroring.  `#bind_values` hasn't existed since Rails 5.2, which means this line of code hasn't been ran in a long time.  With the upgrade to Rails 7.1, we are running into this line of code again, because some `relation`s are in fact `ActiveRecord::Relation`s.